primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -608,6 +608,35 @@ public static boolean isNestedWithinIterator(UIComponent component) {
         return false;
     }
 
+    /**
+     * GitHub #7763 Renders an in memory UIComponent and updates its ID with an index.
+     * For example id="form:test" becomes id="form:test:0".
+     *
+     * @param context the {@link FacesContext}.
+     * @param component the {@link UIComponent} to render.
+     * @param index the index number to append to the ID
+     * @throws IOException if any IO error occurs
+     */
+    public static void encodeIndexedId(FacesContext context, UIComponent component, int index) throws IOException {
+        // swap writers
+        ResponseWriter writer = context.getResponseWriter();
+        FastStringWriter fsw = new FastStringWriter();
+        context.setResponseWriter(writer.cloneWithWriter(fsw));
+
+        // encode the component
+        component.encodeAll(context);
+
+        // restore the original writer
+        context.setResponseWriter(writer);
+
+        // append index to all id's
+        char separator = UINamingContainer.getSeparatorChar(context);
+        String encodedComponent = fsw.toString();
+        encodedComponent = encodedComponent.replaceAll("\\sid=\"(.*?)\"", " id=\"$1" + separator + index + "\"");
```

Comment:
Just to avoid a mindgap. Does `id=\"$1"` already contains the first separator, so you only need to add the second and the index to achieve `id="form:test:0"`

Reply:
yes the capture group captures the entire current ID so can then append `:1`, `:2` etc onto it!